### DISCUSSION TITLE
Dynamic response schema based on query parameters

### DISCUSSION THREAD

**Discussion by nerdstep** [ecyrbe/zodios#461] [Q&A] [answered]
This question is similar to https://github.com/ecyrbe/zodios/discussions/448 but perhaps a little more complex in nature.

We have an API implementing the [JSON:API](https://jsonapi.org/) spec, which allows things like for example [Sparse Fieldsets](https://jsonapi.org/format/#fetching-sparse-fieldsets). In other words, we may have a query like `GET /articles?fields=title,body,date`, where `fields` could be any arbitrary set of field names. So essentially we'd need a way to read the query parameters passed to a callback function on the `response` property in order to then omit any fields from the schema that were not specified in the `fields` list.

Would a feature like this even be doable?

**Comment by ecyrbe** [comment, marked as answer]
Hello,
This would be really hard to implement and i'm not sure DX would be that good to define such complex behaviour.
But any Idea on how to define an API with output tied to input without using a gazillion schemas would  be a first step.
For this kind of feature, we would need to create an RFC first where we could define every aspect on how declare such a beast

**Reply by nerdstep** [reply]
Hey @ecyrbe thanks for the reply. The more I think about it, I'm realizing that there's probably no practical way to even dynamically alter a zod schema like that, e.g. using `pick` or `omit`, which is how I was originally thinking it might be done. One certainly would not want to manually define every possible permutation of the schema, which means you'd probably need to start getting into code generation territory... 

Yeah, juice doesn't really feel worth the squeeze on this one.
